lib/Dialect/FIRRTL/Transforms/LowerTypes.cpp

```diff
@@ -664,6 +665,57 @@ void TypeLoweringVisitor::visitExpr(SubindexOp op) {
   opsToRemove.push_back(op);
 }
 
+static IntegerAttr getIntAttr(const APInt &value, MLIRContext *context) {
+  return IntegerAttr::get(IntegerType::get(context, value.getBitWidth()),
+                          value);
+}
+
+// This generates a mux tree for selecting the corresponding element from the
+// flattened vector based on the index.
+// x = a[index] is lowered to ::
+// x = index == 3 ? a_3 : index == 2 ? a_2 : index == 1 ? a_1:a_0
+// TODO: This currently does not handle vector of bundles, multi-dim vectors and
+// assignment to vectors.
+void TypeLoweringVisitor::visitExpr(SubaccessOp op) {
+  Value input = op.input();
+  Value index = op.index();
+  FIRRTLType resultType = op.getType();
+  FIRRTLType indexType = index.getType().cast<FIRRTLType>();
+  FVectorType inputType;
+  if (auto FlipV = input.getType().cast<FIRRTLType>().dyn_cast<FlipType>()) {
+    inputType = FlipV.getElementType().cast<FVectorType>();
+    builder->emitError(
+        "Lowering of dynamic assignment to vectors not supported");
+    assert("Lowering of dynamic assignment to vectors not supported");
+  } else
+    inputType = input.getType().cast<FIRRTLType>().cast<FVectorType>();
+  if (inputType.getElementType().isa<BundleType>()) {
+    builder->emitError("Lowering of vectors of bundle type not supported");
+    assert("Lowering of vectors of bundle type not supported");
```

Comment:
The assert conditions are always true, which probably isn't what you mean: do you mean `assert(0 &&` to die in the unsupported cases?

Please don't do this :-), please recover gracefully on unsupported thing instead of crashing.